editor/src/clj/editor/defold_project.clj

```diff
@@ -142,40 +146,64 @@
                   batch
                   load-deps)))))))
 
-(defn load-resource-nodes [project node-ids render-progress! resource-node-dependencies]
+(defn- load-resource-nodes [project node-ids render-progress! resource-node-dependencies resource-metrics]
   (let [evaluation-context (g/make-evaluation-context)
         node-id->resource (into {}
                                 (map (fn [node-id]
                                        [node-id (g/node-value node-id :resource evaluation-context)]))
                                 node-ids)
         old-nodes-by-resource-path (g/node-value project :nodes-by-resource-path evaluation-context)
-        loaded-nodes-by-resource-path (into {} (map (fn [[node-id resource]]
-                                                      [(resource/proj-path resource) node-id]))
+        loaded-nodes-by-resource-path (into {}
+                                            (map (fn [[node-id resource]]
+                                                   [(resource/proj-path resource) node-id]))
                                             node-id->resource)
         nodes-by-resource-path (merge old-nodes-by-resource-path loaded-nodes-by-resource-path)
         loaded-nodes (set node-ids)
-        load-deps (fn [node-id] (node-load-dependencies node-id loaded-nodes nodes-by-resource-path resource-node-dependencies evaluation-context))
+        load-deps (fn [node-id]
+                    (du/measuring resource-metrics (resource/proj-path (g/node-value node-id :resource evaluation-context)) :find-new-reload-dependencies
+                      (node-load-dependencies node-id loaded-nodes nodes-by-resource-path resource-node-dependencies evaluation-context)))
         node-ids (sort-nodes-for-loading loaded-nodes load-deps)
         basis (:basis evaluation-context)
         render-loading-progress! (progress/nest-render-progress render-progress! (progress/make "" 5 0) 4)
         render-processing-progress! (progress/nest-render-progress render-progress! (progress/make "" 5 4))
+        resource-metrics-load-timer (du/when-metrics
+                                      (volatile! 0))
+        start-resource-metrics-load-timer! (du/when-metrics
+                                             (fn []
+                                               (vreset! resource-metrics-load-timer (System/nanoTime))))
+        stop-resource-metrics-load-timer! (du/when-metrics
+                                            (fn [resource-path]
+                                              (let [end-time (System/nanoTime)
+                                                    ^long start-time @resource-metrics-load-timer]
+                                                (du/update-metrics resource-metrics resource-path :process-load-tx-data (- end-time start-time)))))
```

Comment:
I think the abstraction for metrics could be a bit cleaner? I like how it's done in `util.profiler`, where you provide a keyword id to the profile block. We could have something similar instead? E.g.
```clj
(du/begin ::load) ;; compiles to nothing if metrics are disabled, 
                  ;; otherwise starts measuring in a global metrics atom
```

Also, I think `(System/nanoTime)` should and `end-start` stuff should be fully encapsulated in this measure util